Automatically sideload from inserted USB drives

mwleeds opened the pull request:
Fixes https://github.com/flatpak/flatpak/issues/3490

mwleeds commented:
Marked this as a draft because I'm not sure I got the automake parts right, but it can still be reviewed.

mwleeds commented:
I guess we will need a configure time option to enable this, since maybe every distribution doesn't want it.

alexlarsson commented:
Shellcheck says:
```
Line 11:
for f in $1/*; do
         ^-- SC2231: Quote expansions in this for loop glob to prevent wordsplitting, e.g. "$dir"/*.txt .
```

alexlarsson commented on sideload-repos-systemd/flatpak-create-sideload-symlinks.sh:
```diff
@@ -0,0 +1,29 @@
+    if ! test -d "$f"; then
+        continue
+    fi
+    base=$(basename "$f")
```
I wonder if just using the basename is enough? Maybe it could append some checksum of the entire path?

mwleeds replied:
Good point. I think we could just use `automount-$(sha256sum $f)`, no need to include the basename I think.

alexlarsson commented on sideload-repos-systemd/flatpak-create-sideload-symlinks.sh:
```diff
@@ -0,0 +1,29 @@
+done
+
+# Remove any broken symlinks e.g. from drives that were removed
+for f in /run/flatpak/sideload-repos/*; do
```
Also, should maybe have a prefix so we don't remove unrelated links here?

mwleeds replied:
I was thinking a broken link probably shouldn't be there anyway, but you're right it would be cleaner to only remove our own stuff. I can use `automount-` as a prefix I suppose.

alexlarsson commented on common/flatpak-dir.c:
```diff
@@ -11906,8 +11907,30 @@ add_subdirs (GPtrArray *res,
+              g_autoptr(GFile) dot_ostree_repo_subpath = g_file_resolve_relative_path (path, ".ostree/repo");
+              g_autoptr(GFile) dot_ostree_repo_d_subpath = g_file_resolve_relative_path (path, ".ostree/repos.d");
+
+              if (ostree_repo_open (ostree_repo_subpath, NULL, NULL))
```
Isn't this calling ostree_repo_open() on a GFile instead of an OstreeRepo. You need a ostree_repo_new() somewhere.

mwleeds replied:
Yes, my mistake

alexlarsson commented:
```
sideload-repos-systemd/Makefile.am.inc:11: error: cannot apply '+=' because 'systemdsystemunit_DATA' is not defined in
sideload-repos-systemd/Makefile.am.inc:11: the following conditions:
sideload-repos-systemd/Makefile.am.inc:11:   !BUILD_SYSTEM_HELPER
sideload-repos-systemd/Makefile.am.inc:11: either define 'systemdsystemunit_DATA' in these conditions, or use
```

mwleeds commented:
> Shellcheck says (for flatpak-create-sideload-symlinks.sh):
> 
> ```
> Line 11:
> for f in $1/*; do
>          ^-- SC2231: Quote expansions in this for loop glob to prevent wordsplitting, e.g. "$dir"/*.txt .
> ```

Fixed this, and also changed it to link to all mounts found regardless of if there's a repo on them, because I think it's confusing to a user if a USB isn't being used because they have to remove and re-insert it.
